darts/utils/statistics.py

```diff
@@ -189,6 +192,19 @@ def extract_trend_and_seasonality(
             **kwargs,
         ).fit()
 
+    elif method == "MSTL":
+        raise_if_not(
+            model in [SeasonalityMode.ADDITIVE, ModelMode.ADDITIVE],
+            f"Only ADDITIVE model is compatible with the MSTL method. Current model is {model}.",
+            logger,
+        )
+
```

Comment:
Also check that freq is a `List` for only if `method == "MSTL"` before the if/else.

Reply:
Implemented and updated testing.

Reply:
Updated and added tests.